### PR TITLE
Bug 1791934: independent controller for clusteroperators

### PR DESCRIPTION
**Pull request by dmage**

### CODE REVIEW

File: pkg/operator/clusteroperator.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	configv1listers "github.com/openshift/client-go/config/listers/config/v1"
+	imageregistryv1informers "github.com/openshift/client-go/imageregistry/informers/externalversions/imageregistry/v1"
+	imageregistryv1listers "github.com/openshift/client-go/imageregistry/listers/imageregistry/v1"
+
+	"github.com/openshift/cluster-image-registry-operator/defaults"
+	regopclient "github.com/openshift/cluster-image-registry-operator/pkg/client"
+	"github.com/openshift/cluster-image-registry-operator/pkg/resource"
+)
+
+const workQueueKey = "instance"
```
Comment (adambkaplan) [inline pkg/operator/clusteroperator.go]: May want to name this const to be more specific. I may have a similar need for a constant queue key in my metrics PR

File: pkg/operator/controller.go [hunk 2]
```diff
@@ -470,13 +463,23 @@ func (c *Controller) Run(stopCh <-chan struct{}) error {
...
+
 	configInformerFactory.Start(stopCh)
 	kubeInformerFactory.Start(stopCh)
 	openshiftConfigKubeInformerFactory.Start(stopCh)
 	kubeSystemKubeInformerFactory.Start(stopCh)
 	regopInformerFactory.Start(stopCh)
 	routeInformerFactory.Start(stopCh)
 
+	// TODO(dmage): This controller should be started from main.
+	go clusterOperatorStatusController.Run(stopCh)
```
Comment (adambkaplan) [inline pkg/operator/controller.go]: Or from a separate "starter" struct that is effectively main. See https://github.com/openshift/cluster-image-registry-operator/pull/447/commits/eb53b988d47e8a8acfc7f5d00b73bb0fbb19b515

Comment (dmage) [reply pkg/operator/controller.go]: Sure. I'd prefer to avoid a struct with one method and have just a function. But it doesn't really matter for me.

**Comment by dmage** [comment]
@adambkaplan this should fix upgrade problems, so we'll need to backport it

**Comment by dmage** [comment]
@adambkaplan teardown issues should be fixed by #452.

**Pull request merged**